File: VB/HeckYaBaby/Backend/Presentation/IHeckYaBabyService.vb
Imports System.ServiceModel
Imports System.ServiceModel.Web

Namespace Presentation

    <ServiceContract>
    Friend Interface IHeckYaBabyService

        <OperationContract>
        <WebInvoke(Method:="OPTIONS", UriTemplate:="*")>
        Sub Options()

        <OperationContract>
        <WebInvoke(Method := "POST", UriTemplate := "/Profile", RequestFormat := WebMessageFormat.Xml, ResponseFormat := WebMessageFormat.Json)>
        Function PostProfile(profile As Profile) As Profile 

        <OperationContract>
        <WebInvoke(Method := "GET", UriTemplate := "/Profile", ResponseFormat := WebMessageFormat.Json)>
        Function GetProfiles() As ProfilePage

        <OperationContract>
        <WebInvoke(Method := "PUT", UriTemplate := "/Profile/{id}", RequestFormat := WebMessageFormat.Xml, ResponseFormat := WebMessageFormat.Json)>
        Function UpdateProfile(id As string, profile As Profile) As Profile 

        <OperationContract>
        <WebInvoke(Method := "DELETE", UriTemplate := "/Profile/{id}", RequestFormat := WebMessageFormat.Xml, ResponseFormat := WebMessageFormat.Json)>
        Sub DeleteProfile(id As String)

    End Interface

End NameSpace

File: C#/HeckYaBaby/Service/IHeckYaBabyService.cs
using System.Collections.Generic;
using System.ServiceModel;
using System.ServiceModel.Web;

namespace Service
{
    [ServiceContract]
    public interface IHeckYaBabyService
    {
        [OperationContract]
        [WebInvoke(Method = "GET", UriTemplate = "/Profile", ResponseFormat = WebMessageFormat.Json)]
        IEnumerable<Profile> GetProfile();

        [OperationContract]
        [WebInvoke(Method = "OPTIONS", UriTemplate = "*")]
        void Options();

        [OperationContract]
        [WebInvoke(Method = "PUT", UriTemplate = "/Profile/{id}", RequestFormat = WebMessageFormat.Xml, ResponseFormat = WebMessageFormat.Json)]
        Profile UpdateProfile(string id, Profile profile);

        [OperationContract]
        [WebInvoke(Method = "POST", UriTemplate = "/Profile", RequestFormat = WebMessageFormat.Xml, ResponseFormat = WebMessageFormat.Json)]
        Profile PostProfile(Profile profile);

        [OperationContract]
        [WebInvoke(Method = "DELETE", UriTemplate = "/Profile/{id}", RequestFormat = WebMessageFormat.Xml, ResponseFormat = WebMessageFormat.Json)]
        void DeleteProfile(string id);
    }
}
